Implementing video to the medium creation use case

sebastien-deren opened the pull request:
better separation of classes (listener only listen and dispatch) 
two Media services:

-  one to manage doctrine entity
- the other to manage File upload and delete

sebastien-deren committed:
Creation of a basic CRUD for figures.
Securisation of certain feature via twig, template for message implementation
Slug created dynamically to be unique
/!\ Need to migrate doctrine to include Unique constraint on name

sebastien-deren committed:
Modification to show fixtures category (ENUM not implemented in fixtures)
Commenting the User AUTHENTICATED code cause user's use case is not implemented yet

sebastien-deren committed:
correcting conflicting files

sebastien-deren committed:
correcting conflicting files

sebastien-deren committed:
correcting conflicting files

sebastien-deren committed:
correcting conflicting files

sebastien-deren committed:
correcting conflicting files

sebastien-deren committed:
correcting conflicting files

sebastien-deren committed:
disabling messages

sebastien-deren committed:
Implementation of the Create and edit Media use case
Creation of an embedded dynamic form
creation of an entitylistener to upload the file on persist
the Edit use case need rework (unlink not working) will be seen on the delete use case
EDIT:IT worked but the path was not correct.

sebastien-deren committed:
Some cleaning, and removing unused methods/

sebastien-deren committed:
Implementing video to the medium creation use case
better separation of classes (listener only listen and dispatch)
two Media services one to manage doctrine entity
The other to manage File upload and delete

sebastien-deren committed:
adding checks to our setters

sebastien-deren committed:
adding a makefile to facilitate the CLI

sebastien-deren committed:
correcting conflicting files

sebastien-deren committed:
moving js to assets for better clarity

sebastien-deren committed:
correcting conflicting files

sebastien-deren committed:
Creation of a choicefield
formlistener to check which type of medium is upload
we will use JS to show the selected field (and hide the other one)

sebastien-deren committed:
deleting media form

sebastien-deren committed:
correcting with symfony insight

sebastien-deren committed:
Creation of our readme
some minor change to assure good installation

sebastien-deren committed:
minor makefile change

Lionelsan95 commented on src/Service/Media/MediaFileManager.php:
```diff
@@ -0,0 +1,33 @@
+            return;
+        }
+        try {
+                unlink($filePath) ?? throw new \Exception($filePath);
```
```suggestion
                if(false === unlink($filePath)){
                    throw new \Exception($filePath);
                }
```
si la suppression échoue, on renvoi **false** et non null

Lionelsan95 commented on src/Service/Media/MediaFileManager.php:
```diff
@@ -0,0 +1,33 @@
+        }
+        try {
+                unlink($filePath) ?? throw new \Exception($filePath);
+        } catch (\Exception $exception) {
+            $this->logger->error('the media at: {filePath} couldn\'t be removed', [
+                'filePath' => $filePath,
+            ]);
+        }
+    }
```
Pas nécessaire que tu le fasse ici mais en général, le service renvoi une exception sans la catch. 

C'est la personne appel ce service qui le fera et mettra des logs s'ils veut. 

Single responsability :). Son boulout c'est juste de supprime et si je ne peux pas Je renvoi une alerte et je m'arrête.

Lionelsan95 reviewed (commented):
done

sebastien-deren committed:
Feature/implement delete medium#17 (#32)

* Making the medium deletable (already deletable by Type allow_delete)
Adding the Js function to delete the Medium

* Adding a Media controller to update create delete media
Reading is done in the figure controller we don't need to read it in other way
moving MediaListener in EventSubscriber folder.

* adding user authentication to Media CRUD

* creation of a carrousel with multiple element

* Adding Carousel to figure Homepage
some problem with carousel animation, but that's like 100% frontend


* Add a bit of js to make the media choice type dynamic (#41)

It will only show the field selected
On the back-end we make sure to strip of value the not selected field.
It'll make sure the user has what he want


* Implement Message Pagination (#36)

Creation of a js script for paginate the comments.
(creation of a paginator js module to be reused if needed)
modifying the displayMessage method to return an array chucked by pages.

sebastien-deren committed:
correction based on lione owono review

sebastien-deren committed:
correcting conflicting files

sebastien-deren committed:
Merge branch 'develop' into feature/implement_CreateVideo
